dev_tools/requirements/re-pip-compile.sh

```diff
@@ -17,15 +17,16 @@
 set -ex
 
 # First, compile a complete & consistent environment with all dependencies
-pip-compile --upgrade --output-file=dev.env.txt --resolver=backtracking deps/dev-tools.txt deps/runtime.txt
+mkdir -p ./envs/
+pip-compile --output-file=envs/dev.env.txt --resolver=backtracking deps/dev-tools.txt deps/runtime.txt
 
 # Compile the various sub-environments.
 # For each of these, we include `constrain.txt` as an input requirements file. This file
-# has only one line: "-c ../dev.env.txt", which will use the above rendered dev.env.txt as a
+# has only one line: "-c envs/dev.env.txt", which will use the above rendered dev.env.txt as a
 # "constraints file", which is exactly what we want.
-pip-compile --output-file=runtime.env.txt --resolver=backtracking constrain.txt deps/runtime.txt
-pip-compile --output-file=format.env.txt  --resolver=backtracking constrain.txt deps/runtime.txt deps/format.txt
-pip-compile --output-file=pylint.env.txt  --resolver=backtracking constrain.txt deps/runtime.txt deps/pylint.txt
-pip-compile --output-file=pytest.env.txt  --resolver=backtracking constrain.txt deps/runtime.txt deps/pytest.txt
-pip-compile --output-file=docs.env.txt    --resolver=backtracking constrain.txt deps/runtime.txt deps/docs.txt
-
+pip-compile --output-file=envs/runtime.env.txt   --resolver=backtracking constrain.txt deps/runtime.txt
```

Comment:
```
for pip_env in runtime format pylint pytest docs pip-tools; do 
  pip-compile --output-file "envs/${pip_env}.env.txt" --resolver backtracing constrain.txt deps/runtime.txt "deps/${pip_env}.txt
  done
  ```
  WDYT?

Reply:
My personal rule is "never write a loop in a bash script"

The idea being once you do that you're on the path of no return to an incomprehensible mess of bash when you probably should have switched to a real ™️  scripting language. I'll keep this unrolled for now, but will keep in mind going forward 

Also I will point out that we don't want `runtime.txt` included in the pip-tools env